Vizzuality/marxan-cloud
# Add the scenario (“BLM”) chart

### Overview

This PR adds a new component, `BlmChart`, that corresponds to the chart displayed when the user is creating a scenario.

<p align="center">
<img src="https://user-images.githubusercontent.com/6073968/108734384-b6758580-752f-11eb-859d-2ecbed895f4b.png" alt="Line chart with thumbnails attached to some points" />
</p>

When the user hovers a point, only that point has its thumbnail shown. If some points don't have a thumbnail, hovering them will change nothing.

By default, the thumbnails of the first and last point, as well as the one of the “BLM” point are shown. Similarly to above, if one of these points doesn't have a thumbnail, it will not be shown by default.

### Designs

The chart can be found on [this screen](https://www.figma.com/file/hq0BZNB9fzyFSbEUgQIHdK/Marxan-Visual_V02?node-id=1523%3A3915) on Figma.

There are two elements of the design that I haven't implemented:
1. The blue area below the lines should have rounded corners. I'm unsure how to implement this.
2. The same with the lines that connect the points to their thumbnails: they should have rounded corners.

I'd like to point out that the algorithm that positions the thumbnails assumes the points draw a curve similar to `f: x => 1 / x`, when `x > 0`. If the curve is different, the thumbnails could be displayed on top of each other, on top of the line or even outside the container.

I'm also assuming the points will be fairly distributed along the curve. If they are not, the same could happen.

There's another edge case that could compromise the chart visually: if the BLM point is too high, the text “Recommended (BLM)” could be on top of “Boundary length”, as it is absolutely positioned.

### Testing instructions

Here's what you can test:
1. The chart visually complies with the design
2. If you add or remove the thumbnail of points, when hovering them, the behaviour should be the one described above
3. The chart should respond to changes in width/height with grace
